server/src/main/java/org/opensearch/threadpool/ThreadPoolStats.java

```diff
@@ -146,6 +160,10 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             if (completed != -1) {
                 builder.field(Fields.COMPLETED, completed);
             }
+            if (waitTimeNanos != -1) {
+                // wait time is tracked in nanos but TimeValue raw value is in millis
+                builder.humanReadableField(Fields.WAIT_TIME_MILLIS, Fields.WAIT_TIME, getWaitTimeNanos());
```

Comment:
I was looking at the `search.query_time` metric which also uses this `TimeValue` object:

https://github.com/opensearch-project/OpenSearch/blob/56a87362c728f4d6c13f74eaacea8c7e50ed6a83/server/src/main/java/org/opensearch/index/search/stats/SearchStats.java#L229-L231

That being said, if we want to maintain `nanos` precision, then we do need to do what you're suggesting above because the way `TimeValue` is currently implemented only supports `millis` precision:

https://github.com/opensearch-project/OpenSearch/blob/bf10ff71029e456f4ef6e0ef4015afe056f5240b/server/src/main/java/org/opensearch/common/xcontent/XContentOpenSearchExtension.java#L119-L125

From the user perspective, I do think having a human readable field is more useful than having the additional precision of maintaining `nanos` precision compared to rounding to `milli` since I don't think anyone would be concerned about this metric while it is on the order of nanoseconds. Also there's going to be some amount of rounding no matter how precise you measure the time anyways.

Reply:
Ah apologies just realized you were suggesting we have the nanos as the raw value but keep the human readable field -- that works for me, but let me see if there's a way to implement this in the `TimeValue` or `builder` classes itself instead of using this if statement.

Reply:
I don't see a straightforward way of doing this in the `builder` class or the `TimeValue` class so I went ahead and implemented it like you suggested with the if statement.